Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector.py

```diff
@@ -0,0 +1,264 @@
+            params=params
+        )
+        return data['objects']
+
+
+def get_jwt_expiration(token: str):
+    if "." not in token:
+        return 0
+    jwt_token = base64.b64decode(token.split(".")[1] + '==')
+    jwt_token = json.loads(jwt_token)
+    return jwt_token.get("exp")
+
+
+def generate_new_token(self):
+    token = self.authenticate()
+    self.save_jwt_token(token)
+    self._save_access_token(token)
+
+
+def test_module(client: Client, params: Dict[str, Any]) -> str:
+    """
+    Tests API connectivity and authentication'
+    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
+    successful.
+    Raises exceptions if something goes wrong.
+    Args:
+        client (Client): Gurdicore client to use.
+        params (Dict): Integration parameters.
+    Returns:
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+
+    try:
+        client.login()
+
+    except Exception as e:
+        if 'UNAUTHORIZED' in str(e):
+            return 'Authorization Error: make sure the username and password are correctly set'
+        else:
+            raise e
+
+    return 'ok'
+
+
+def add_time_to_events(events):
+    """ Adds the _time key to the events.
+
+    Args:
+        events: list[dict] - list of events to add the _time key to.
+
+    Returns:
+        list: The events with the _time key.
+    """
+    if events:
+        for event in events:
+            event['_time'] = event.get('start_time')
+
+
+def get_events(client, args):
+    """
+       Gets events from Guardicore API.
+    """
+    start_time = date_to_timestamp(
+        arg_to_datetime(args.get('from_date', '1 second ago')))
+    end_time = date_to_timestamp(
+        arg_to_datetime(args.get('to_time'))) if "to_time" in args else int(datetime.now().timestamp() * 1000)
+    limit = arg_to_number(args.get("limit", 1000))
+    offset = int(args.get('offset', 0))
+    events = client.get_events(start_time, end_time, limit, offset)
+    hr = tableToMarkdown(name='Test Event', t=events)
+    return events, CommandResults(readable_output=hr)
+
+
+def fetch_events(client: Client, params: dict, last_run: dict) -> List[dict]:
+    """
+       Fetches events from Guardicore API.
+    """
+    start_time = date_to_timestamp(arg_to_datetime(last_run.get('from_ts', '1 second')))
+    end_time = int(datetime.now().timestamp() * 1000)
+    demisto.debug(f'Getting events from: {timestamp_to_datestring(start_time)}, till: {timestamp_to_datestring(end_time)}')
+    offset = arg_to_number(last_run.get('offset')) or 0
+    limit = arg_to_number(params.get('max_events_per_fetch')) or 1000
+
+    retrieve_events = client.get_events(start_time, end_time, limit, offset)
+    demisto.debug(f'Fetched {len(retrieve_events)} events.')
+
+    demisto.setLastRun(
+        {'from_ts': end_time} if len(retrieve_events) < limit else {'from_ts': start_time, 'offset': offset + limit}
```

Comment:
Does the API treat `start_time` as `gt` or `gte`? i.e. will there be a need to dedupe the data between 2 subsequent runs?